src/commonAccountFunctions.ts

```diff
@@ -225,3 +255,40 @@ export async function ensureDataset<T extends Account>(this: T, name: string, ne
   }
   return this.addDataset(name, newDs);
 }
+
+export async function ensureQuery<T extends Account>(
+  this: T,
+  name: string,
+  newQuery: NewQueryWithService | NewQueryWithDataset
+) {
+  try {
+    const query = await this.getQuery(name);
+    const info = await query.getInfo();
+    if (newQuery?.accessLevel && newQuery?.accessLevel !== info.accessLevel) {
+      throw getErr(
+        `Query ${name} already exists with access level ${info.accessLevel}. Cannot ensure it with access level ${newQuery?.accessLevel}. Please change the access level to match the Query, or remove it entirely as a parameter.`
+      );
+    }
+    return query;
+  } catch (e: any) {
+    if (e.statusCode !== 404) throw e;
+  }
+  // Cast here to keep typescript happy
+  return this.addQuery(name, newQuery as NewQueryWithDataset);
```

Comment:
similar to a previous comment, from a readability perspective I'd argue it's better to move this to the catch cluse